rest_framework/utils/field_mapping.py

```diff
@@ -265,6 +265,8 @@ def get_relation_kwargs(field_name, relation_info):
             kwargs.pop('queryset', None)
         if model_field.null:
             kwargs['allow_null'] = True
+        if isinstance(model_field.target_field, models.UUIDField):
```

Comment:
I find it a bit awkward to introduce special handling for one field type while everything else is basically ignored. Sure, PKs are for the most part either `AutoField` or `UUIDField`, but that that is not a sure thing. For arguments sake this might very well be a `DateTimeField`. 

Having more specific types seems reasonable, but then it should be done consistently and across the board imho.

Reply:
Agreed, I was only thinking of integer, string, and UUID keys at the time but I can definitely see the value in supporting more fields.

Is there any reason we wouldn’t want to use `ModelSerializer.serializer_field_mapping` here?

Reply:
This is just something that I noticed and thought it would be worth mentioning. Someone more knowledgeable on these parts should probably chime in on that question. @auvipy?

Reply:
my thought here is, we should initially consider prospective primary key types / fields mapping here initiallly

Reply:
I meant most commonly used primary key types checking / special could be done initially

Reply:
> Is there any reason we wouldn’t want to use `ModelSerializer.serializer_field_mapping` here?

We can use / try it here as well